### PR TITLE
added a git ref in cli reference

### PR DESCRIPTION
**Pull request by girishramnani**
## What is the purpose of this change? What does it change?

added an example of `ref` in cli reference

## Was the change discussed in an issue?

No issue

## How to test changes?

Doc change

### CODE REVIEW

File: docs/cli-reference.md [hunk 1]
```diff
@@ -212,6 +212,9 @@ Catalog related operations
   # Create new Node.js component with source from remote git repository.
   odo create nodejs --git https://github.com/openshift/nodejs-ex.git
 
+  # Create a new Node.js component with source from remote git repository and a branch.
```
[the remark this answers is not in the document]
Comment (girishramnani) [reply docs/cli-reference.md]: so should I add example for that or just explain here `Create a new Node.js component with source from remote git repository using a branch, tag or commit ref`

[the remark this answers is not in the document]
Comment (metacosm) [reply docs/cli-reference.md]: Second option looks good.

[the remark this answers is not in the document]
Comment (girishramnani) [reply docs/cli-reference.md]: ack

[the remark this answers is not in the document]
Comment (girishramnani) [reply docs/cli-reference.md]: resolved

File: docs/cli-reference.md [hunk 2]
```diff
@@ -212,6 +212,9 @@ Catalog related operations
   # Create new Node.js component with source from remote git repository.
   odo create nodejs --git https://github.com/openshift/nodejs-ex.git
 
+  # Create a new Node.js component with source from remote git repository using a branch, tag or commit ref.
+  odo create nodejs --git https://github.com/openshift/nodejs-ex.git --ref master
```
Comment (anmolbabu) [inline docs/cli-reference.md]: If I remember correctly, this is generated from command help so could you please add it there and generate it to this doc? But let me just confirm the same

Comment (metacosm) [reply docs/cli-reference.md]: Ah yes, forgot about that. Also, I'd use an example with a different `ref` than `master` since that's the default.

Comment (surajnarwade) [reply docs/cli-reference.md]: Please autogenerate it

**Comment by girishramnani** [comment]
should probably close this then and @cdrage being incharge of docs will probably generate it soon

**Comment by girishramnani** [comment]
> > should probably close this then and @cdrage being incharge of docs will probably generate it soon
> 
> The change is fine in itself. It should just happen in the command itself as opposed to the generated file.

Its already there in the CLI examples section

**Comment by cdrage** [comment]
@girishramnani 

If you run `make generate-cli-reference` it will automatically generate the correct documentation :+1: 

Just update the PR with the change!

File: docs/cli-reference.md (hunk 2 above)
Comment (girishramnani) [reply docs/cli-reference.md]: ack, generated. also @metacosm  the nodejs-ex doesn't have a branch that is working hence I added an example for release `v1.0.1`

**Pull request merged**